sounddevice.py

```diff
@@ -2262,13 +2262,68 @@ def reset(self):
         vars(self).clear()
         self.__init__()
 
+
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+
+    Parameters
+    ----------
+    msg : str, optional
+        User error message.
+    code : int, optional
+        PaErrorCode value or None if there is no PaErrorCode associated
+        with the error.
+
+    Notes
+    -----
+    - The `host_api`, `host_code`, and `host_text` attributes will be None
+      except for the case when `code` == `paUnanticipatedHostError`.
+    
+    Attributes
+    ----------
```

Comment:
I did some awkward manipulations in `doc/index.rst`, you'd have to add the `:members:` option:

```rst
.. autoclass:: PortAudioError
   :members:
```

However, the situation has changed since we were discussing this. I'm not sure anymore if it makes more sense to document the class variables instead of listing the attributes in the main docstring as you had done initially. What do you think?